script/verification/HoloceneSystemConfigUpgrade.s.sol

```diff
@@ -0,0 +1,279 @@
+            // everything else.
+            // See https://specs.optimism.io/protocol/system-config.html?highlight=ecotone%20scalar#ecotone-scalar-overhead-uint256uint256-change
+            require(previousScalarEncodingVersion == 0, "scalar-101 previous scalar version != 0 or 1");
+            require(newScalarEncodingVersion == 1, "scalar-102 reenconded scalar version != 1");
+            require(sysCfg.blobbasefeeScalar() == uint32(0), "scalar-103 blobbasefeeScalar !=0");
+
+            // The scalars should match if we add the new scalar version byte to the previous scalar.
+            require(reencodedScalar == previousScalar + (uint256(0x01) << 248), "scalar-104 scalar mismatch");
+        }
+    }
+
+    // Checks the disputeGameFactory address is set correctly after the upgrade
+    function checkDGF() internal view {
+        require(sysCfg.disputeGameFactory() == targetDGF, "scalar-106");
+    }
+
+    // Checks gasPayingToken and its decimals are set correctly after the upgrade
+    function checkGasPayingToken() internal view {
+        (address t, uint8 d) = sysCfg.gasPayingToken();
+        require(t == targetGasPayingToken, "scalar-107");
+        require(d == targetDecimals, "scalar-108");
+    }
+
+    // Checks the remaining storage variables are unchanged after the upgrade
+    function checkBaseSysCfgVars() internal view {
+        // Check remaining storage variables didn't change
+        require(keccak256(abi.encode(getBaseSysCfgVars())) == keccak256(abi.encode(previous)), "system-config-100");
+    }
+
+    // Reads the semantic version of the SystemConfig contract
+    function getSysCfgVersion() internal view returns (string memory) {
+        return sysCfg.version();
+    }
+
+    // Reads the base storage variables of the SystemConfig contract
+    // (the ones which are common to all versions and should not change during the upgrade)
+    function getBaseSysCfgVars() internal view returns (BaseSysCfgVars memory) {
+        return BaseSysCfgVars({
+            owner: sysCfg.owner(),
+            batcherHash: sysCfg.batcherHash(),
+            gasLimit: sysCfg.gasLimit(),
+            unsafeBlockSigner: sysCfg.unsafeBlockSigner(),
+            resourceConfig: sysCfg.resourceConfig(),
+            batchInbox: sysCfg.batchInbox(),
+            l1CrossDomainMessenger: sysCfg.l1CrossDomainMessenger(),
+            l1StandardBridge: sysCfg.l1StandardBridge(),
+            l1ERC721Bridge: sysCfg.l1ERC721Bridge(),
+            optimismPortal: sysCfg.optimismPortal(),
+            optimismMintableERC20Factory: sysCfg.optimismMintableERC20Factory()
+        });
+    }
+}
+
+interface IResourceMetering {
+    struct ResourceConfig {
+        uint32 maxResourceLimit;
+        uint8 elasticityMultiplier;
+        uint8 baseFeeMaxChangeDenominator;
+        uint32 minimumBaseFee;
+        uint32 systemTxMaxGas;
+        uint128 maximumBaseFee;
+    }
+}
+
+interface ISystemConfig {
```

Comment:
Isn't the interface available from the imported `@eth-optimism-bedrock` package at `@eth-optimism-bedrock/src/L1/interfaces/ISystemConfig` or so? Edit: doesn't seem so. But it would be in a more recent commit of the monorepo. E.g. https://github.com/ethereum-optimism/optimism/blob/proposal/op-contracts/v1.8.0/packages/contracts-bedrock/src/L1/interfaces/ISystemConfig.sol has it. On `develop` the `interfaces` got moved into a directory tree next to the `src` tree, if we want to use a recent monorepo commit instead...

Reply:
I actually think I don't want to float this dependency. This task is for a very specific SystemConfig interface so I'm pretty happy with hardcoding the interface here.

Reply:
OK makes sense.